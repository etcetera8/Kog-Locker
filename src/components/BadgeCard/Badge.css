.badge-card-inner {
  display: flex;
  flex-direction: row; }

.badge {
  height: 75px;
  width: 75px;
  position: relative;
  border-radius: 100%;
  border: 1px solid #fff;
  overflow: hidden; }
  .badge .badge-icon {
    color: #fff;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    margin: auto;
    text-align: center;
    z-index: 100;
    width: 65%;
    height: auto; }
  .badge .badge-inner {
    position: absolute;
    right: 0;
    bottom: 0;
    left: 0;
    background: #ff6400; }
  .badge .logo {
    z-index: 1;
    width: 400%;
    height: auto; }

.type {
  color: #fff; }
