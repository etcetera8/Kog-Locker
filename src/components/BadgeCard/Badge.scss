$orange: rgba(255,100,0,1);
.badge-main-container {
  align-items: flex-start;
  justify-content: flex-start;
}

.badge-card {
  max-width: 260px;
}

.full-badge-wrapper {
  display: inline-block;
  position: relative;
  margin: 25px;
  width: 80px;
  text-align: center;
}

.badge-card-inner {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around; 
}

.badge {
  position: relative;
  margin: 0 auto;
  height: 80px;
  width:  80px;
  overflow: hidden;
  border-radius: 100%;
  border: 2px solid #fff;

  .badge-icon {
    position: absolute;
    z-index: 10;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    margin: auto;
    width:65%;
    height: auto;
    text-align: center;
    color: #fff;
  }
  
  .badge-inner {
    position: absolute;
    right: 0;
    bottom: 0;
    left: 0;
    background: $orange;
  }

  .logo {
    z-index: 1;
    width: 400%;
    height: auto;
  }
}

.type {
  margin: 0;
  color: #fff ;
}

.info {
  display: none;
  width: 100px;
  height: 100px;
  border-radius: 50%;
  color: #fff;
}

.badge:hover~.info {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: block;
  padding: 10px;
  z-index: 2;
  border: 2px solid #fff;
  background: $orange;
}

.badge:hover :nth-child(1) {
  width: 90%;
  opacity: 0;
}

.details {
  display: block;
  margin: 5px;
  text-align: left;
  font-size: .8em;
}