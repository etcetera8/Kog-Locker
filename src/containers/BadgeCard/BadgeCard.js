import React from 'react';

const BadgeCard = () => {
  return (
    <section className="badge-card card">
      I'm a badge card
    </section>
  )
}

export default BadgeCard;