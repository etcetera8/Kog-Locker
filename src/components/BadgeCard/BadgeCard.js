import React from 'react';
import Badge from '../Badge/Badge';
import './Badge.css';

const BadgeCard = ({currentEl, goalEl, goalDist, yearStats}) => {
  return (
    <section className="badge-card card">
      <h2 className='title'> Badges </h2>
      <div className='badge-card-inner'>
        <Badge 
          yearStats={yearStats}
          currentEl={currentEl}
          goalEl={goalEl}
          goalDist={goalDist}
        />
      </div>
    </section>
  );
};

export default BadgeCard;