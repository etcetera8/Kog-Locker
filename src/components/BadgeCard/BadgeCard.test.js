import React from 'react';
import { shallow } from 'enzyme';
import BadgeCard from './BadgeCard';

describe('BadgeCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<BadgeCard />)
    expect(wrapper).toMatchSnapshot();
  })
})