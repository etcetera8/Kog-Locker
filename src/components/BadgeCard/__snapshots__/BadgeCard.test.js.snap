// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BadgeCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
           Badges 
</h2>,
        <div
          className="badge-card-inner"
>
          <Connect(Badge) />
</div>,
      ],
      "className": "badge-card card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Badges ",
          "className": "title",
        },
        "ref": null,
        "rendered": " Badges ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Badge) />,
          "className": "badge-card-inner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
             Badges 
</h2>,
          <div
            className="badge-card-inner"
>
            <Connect(Badge) />
</div>,
        ],
        "className": "badge-card card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Badges ",
            "className": "title",
          },
          "ref": null,
          "rendered": " Badges ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Badge) />,
            "className": "badge-card-inner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
